app/controllers/internal_change_notes_controller.rb

```diff
@@ -0,0 +1,16 @@
+class InternalChangeNotesController < ApplicationController
+  def create
+    InternalChangeNote.create(internal_change_note_params.merge(step_by_step_page_id: step_by_step_page.id, created_at: Time.now, author: current_user.name))
```

Comment:
Following on from the comment in the view, you are doing a `new` and a `create` for `InternalChangeNote`. Can you create one object and just save that when the form is submitted.